lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -254,6 +255,16 @@ function collateResults(localConsole, actual, expected) {
     return makeComparison(auditName + ' audit', actualResult, expectedResult);
   });
 
+  /** @type {Comparison[]} */
+  const requestCountAssertion = [];
+  if (expected.networkRequests) {
+    requestCountAssertion.push(makeComparison(
+      'Requests',
+      server.getRequestUrls(),
```

Comment:
it seems like this should be run outside of the assert module and the result passed in, like the other parts of `actual`. `smokehouse.js` is a natural place to do that, but that wouldn't work for the bundled version run for smokerider (which doesn't use `static-server`). Almost needs to be part of `SmokehouseOptions`, similar to the `lighthouseRunner` callback?